model/websocket_client.go

```diff
@@ -104,16 +134,38 @@ func (wsc *WebSocketClient) ConnectWithDialer(dialer *websocket.Dialer) *AppErro
 }
 
 func (wsc *WebSocketClient) Close() {
+	if wsc.closed {
+		return
+	}
+	wsc.quitWriterChan <- struct{}{}
+	close(wsc.writeChan)
```

Comment:
Is there a reason for actually sending data through `quitWriterChan` instead of closing it since we simply use it as a signaling mechanism?
Also, wouldn't checking for the closure of `writeChan` be enough to stop the writer?

Reply:
> Is there a reason for actually sending data through quitWriterChan instead of closing it since we simply use it as a signaling mechanism?

Yeah .. that was primarily the issue in the previous PR. We can close the channel and move forward even if it is still executing the case for `writeChan`. Sending a message actually ensures that the quit case has hit, and it has returned.

> Also, wouldn't checking for the closure of writeChan be enough to stop the writer?

That has the same problem that we wouldn't actually know that the writer has returned. It will return eventually. But then the `conn.Close` might get executed before a `WriteJSON`/`WriteMessage` actually happens.